tasks/rh-sign-cosign-image/rh-sign-cosign-image.yaml

```diff
@@ -0,0 +1,119 @@
+    tekton.dev/tags: release
+spec:
+  description: >-
+    Task to sign container images in a Snapshot spec using the cosign.
+  params:
+    - name: snapshotPath
+      description: Path to the JSON string of the mapped Snapshot spec in the data workspace
+      type: string
+      default: "snapshot_spec.json"
+    - name: dataPath
+      description: Path to the JSON string of the merged data to use in the data workspace
+      type: string
+      default: "data.json"
+  workspaces:
+    - name: data
+      description: workspace to read and save files
+  steps:
+    - name: sign-image
+      image: quay.io/redhat-appstudio/release-service-utils:447ea0580a2cdd48b4091e1df86fab5c3f86d01c
+      script: |
+        #!/usr/bin/env bash
+        if [ "$TEST" != "1" ]; then
+          function kubectl_get_arg_from_config_map() {
+            kubectl get configmap $1 -o jsonpath='{.data.$2}'
+          }
+          function run_inspect() {
+            skopeo inspect --raw $1
+          }
+          function run_cosign () {
+            cosign $@
+          }
+        fi
+        function fetch_cosign_key() {
+          echo $(kubectl_get_arg_from_config_map $1 SIG_KEY)
+        } 
+        function fetch_aws_region() {
+          echo $(kubectl_get_arg_from_config_map $1 AWS_REGION)
+        } 
+        function fetch_aws_access_key() {
+          echo $(kubectl_get_arg_from_config_map $1 AWS_ACCESS_KEY)
+        } 
+        function fetch_aws_secret_access_key() {
+          echo $(kubectl_get_arg_from_config_map $1 AWS_SECRET_ACCESS_KEY)
+        } 
+        function fetch_registry_user() {
+          echo $(kubectl_get_arg_from_config_map $1 REGISTRY_USER)
+        } 
+        function fetch_registry_password() {
+          echo $(kubectl_get_arg_from_config_map $1 REGISTRY_PASSWORD)
+        }
+
+        SNAPSHOT_PATH=$(workspaces.data.path)/$(params.snapshotPath)
+
+        DATA_FILE="$(workspaces.data.path)/$(params.dataPath)"
+        if [ ! -f "${DATA_FILE}" ] ; then
+            echo "No valid data file was provided."
+            exit 1
+        fi
+
+        request=$(jq -r '.sign.request // "hacbs-signing-pipeline"' ${DATA_FILE})
+        CONFIG_MAP_NAME=$(jq -rc '.sign.configMapName // "signing-config-map"' ${DATA_FILE})
+
+        COMPONENTS_LENGTH=$(jq '.components |length' ${SNAPSHOT_PATH})
+
+        COSIGN_SIG_KEY=$(fetch_cosign_key $CONFIG_MAP_NAME)
+        AWS_REGION=$(fetch_aws_region $CONFIG_MAP_NAME)
+        AWS_ACCESS_KEY=$(fetch_aws_access_key $CONFIG_MAP_NAME)
+        AWS_SECRET_ACCESS_KEY=$(fetch_aws_secret_access_key $CONFIG_MAP_NAME)
+        COSIGN_REGISTRY_USER=$(fetch_registry_user $CONFIG_MAP_NAME)
+        COSIGN_REGISTRY_PASSWORD=$(fetch_registry_password $CONFIG_MAP_NAME)
+
+        for (( COMPONENTS_INDEX=0; COMPONENTS_INDEX<COMPONENTS_LENGTH; COMPONENTS_INDEX++ )); do
+            PUB_CONTAINER_IMAGE=$(jq -r ".components[${COMPONENTS_INDEX}].containerImage" ${SNAPSHOT_PATH})
```

Comment:
I was using tasks/rh-sign-image/tests/test-rh-sign-image-multiple-components.yaml as source where containerImage seems to be public one. Also in
tasks/rh-sign-image/rh-sign-image.yaml quay.io
reference is replaced with registry.redhat.io

reference=$(jq -r ".components[${COMPONENTS_INDEX}].repository" ${SNAPSHOT_PATH}
reference=${reference/quay.io\/redhat-prod/registry.redhat.io}

Reply:
> I was using tasks/rh-sign-image/tests/test-rh-sign-image-multiple-components.yaml as source where containerImage seems to be public one.

Yeah, unfortunately that looks like a mistake in the test. This same mistake can be seen in the push-snapshot task tests. In other places I see stuff like `"source@mydigest"`. In reality, it will be `"quay.io/redhat-user-workloads/rhtas-tenant/rekor-search-ui/rekor-search-ui@sha256:...."`

Reply:
> Also in tasks/rh-sign-image/rh-sign-image.yaml quay.io reference is replaced with registry.redhat.io

Hmm, that is actually a good point. You're probably familiar with the mapping of quay repos to corresponding registry.redhat.io repos. If not, here's a brief explanation: https://github.com/konflux-ci/release-service-utils/blob/main/pyxis/create_container_image.py#L7

I can see that when you call `cosign`, you also include `--sign-container-identity`. So I wonder which identity should be used. When we sign images with simple signing using RADAS, we sign the public-facing references - registry.redhat.io and registry.access.redhat.com. Do you know if that should be the case here too? Probably, because that's what end users will use? (Customers don't have access to pull from quay.io/redhat-prod directly.)

Reply:
OK, cool, so you will have to use the same approach as rh-sign-image does.

Reply:
I still don't see the translation to registry.redhat.io happening anywhere.

Reply:
fixed

Reply:
OK, thanks. I added another comment there.